src/Mesh/Interpolation.jl

```diff
@@ -0,0 +1,371 @@
+    El::UnitRange{Int64}
+    Nel::Integer
+
+    xbnd::Array{FT}        # domain bounds, [2(min/max),ndim]
+    xres::Array{FT}        # resolutions in x1, x2, x3 directions for the uniform grid
+
+    ξ1::Vector{Vector{FT}} # unique ξ1 coordinates of interpolation points within each element 
+    ξ2::Vector{Vector{FT}} # unique ξ2 coordinates of interpolation points within each element 
+    ξ3::Vector{Vector{FT}} # unique ξ3 coordinates of interpolation points within each element 
+  
+    x::Vector{Array{FT}}   # x[elno][3,npts] -> (x1,x2,x3) coordinates of portion of interpolation grid embedded within each element 
+
+    V::Vector{Vector{FT}}  # interpolated variable within each element
+#--------------------------------------------------------
+function Interpolation_Brick(grid::DiscontinuousSpectralElementGrid, xres::Array{FT}) where FT <: AbstractFloat
+    T = Integer
+    ndim = 3
+    xbnd = zeros(FT, 2, ndim) # domain bounds (min,max) in each dimension
+    for dim in 1:ndim 
+        xbnd[1,dim], xbnd[2,dim] = FT(minimum( grid.topology.elemtocoord[dim,:,:] )), FT(maximum( grid.topology.elemtocoord[dim,:,:] )) 
+    end
+    x1g, x2g, x3g = range(xbnd[1,1], xbnd[2,1], step=xres[1]), range(xbnd[1,2], xbnd[2,2], step=xres[2]), range(xbnd[1,3], xbnd[2,3], step=xres[3]) 
+    #-----------------------------------------------------------------------------------
+    El =  grid.topology.realelems # Element (numbers) on the local processor
+    Nel = length(El)
+    n123  = zeros(T,    ndim)     # # of unique ξ1, ξ2, ξ3 points in each cell
+    xsten = zeros(T, 2, ndim)     # x1, x2, x3 start and end for each brick element
+    xbndl = zeros(T, 2, ndim)     # location of x1,x2,x3 limits (min,max) for each brick element
+
+    ξ1, ξ2, ξ3 = map( i -> zeros(FT,i), zeros(T,Nel)), map( i -> zeros(FT,i), zeros(T,Nel)), map( i -> zeros(FT,i), zeros(T,Nel))
+    V  = map( i -> zeros(FT,i), zeros(T,Nel))
+    x  = map( i -> zeros(FT,ndim,i), zeros(T,Nel)) # interpolation grid points embedded in each cell 
+    #-----------------------------------------------------------------------------------
+    for el in 1:Nel
+        for (ξ,xg,dim) in zip((ξ1,ξ2,ξ3), (x1g, x2g, x3g), 1:ndim)
+            xbndl[1,dim], xbndl[2,dim] = minimum( grid.topology.elemtocoord[dim,:,el] ), maximum( grid.topology.elemtocoord[dim,:,el] )
+            xsten[1,dim], xsten[2,dim] = findfirst( temp -> temp ≥ xbndl[1,dim], xg ), findlast( temp -> temp ≤ xbndl[2,dim], xg )
+            n123[dim] = xsten[2,dim] - xsten[1,dim]
+            ξ[el] = Array{FT}(undef,n123[dim])
+            [ ξ[el][i] = 2.0 * ( xg[ xsten[1,dim] + i - 1] - xbndl[1,dim] ) / (xbndl[2,dim]-xbndl[1,dim]) -  1.0 for i in 1:n123[dim] ]
+        end
+
+        x[el] = zeros(FT,ndim,prod(n123))
+        V[el] = zeros(FT,prod(n123))
+
+        ctr = 1
+
+        for k in 1:n123[3], j in 1:n123[2], i in 1:n123[1]
+            x[el][1,ctr] = x1g[ xsten[1,1] + i - 1 ]
```

Comment:
it's not a huge deal, but this does add an extra layer of indirection. I would suggest defining
```julia
x_el = zeros(FT,ndim,prod(n123))
```
before the loop, do 
```suggestion
            x_el[1,ctr] = x1g[ xsten[1,1] + i - 1 ]
```
etc. in the loop, and
```julia
x[el] = x_el
```
outside the loop